src/odemis/gui/cont/tabs/tab_bar_controller.py

```diff
@@ -26,26 +26,28 @@
 import logging
 import wx
 
+from odemis import model
 from odemis.gui.util.wx_adapter import fix_static_text_clipping
 
 from odemis.gui.util import call_in_wx_main
 
 
-class TabBarController(object):
-    def __init__(self, tab_defs, main_frame, main_data):
+class TabController(object):
+    def __init__(self, tab_list, main_frame, main_data, default_tab, use_main_data_tab=False):
         """
-        tab_defs (dict of four entries string -> value):
-           name -> string: internal name
-           controller -> Tab class: class controlling the tab
-           button -> Button: tab btn
-           panel -> Panel: tab panel
+        :param: tab_list: (List[Tab]) list of odemis.gui.cont.tabs.tab.Tab objects.
+        :param: main_frame: odemis.gui.main_xrc.xrcfr_main
+        :param: main_data: (MainGUIData) the main GUI data.
+        :param: default_tab: (Tab) the default_tab to be shown.
+        :param: use_main_data_tab: (bool) flag to use the MainGUIData tab VAEnumerated.
         """
         self.main_frame = main_frame
-        self._tabs = main_data.tab  # VA that we take care of
         self.main_data = main_data
-
-        # create all the tabs that fit the microscope role
-        tab_list, default_tab = self._create_needed_tabs(tab_defs, main_frame, main_data)
+        if use_main_data_tab:
+            self._tab = main_data.tab
+        else:
+            # Current tab (+ all available tabs in choices as a dict tab -> name)
+            self._tab = model.VAEnumerated(None, choices={None: ""})
```

Comment:
You're hidding the state inside the controller. Please take a VA as argument. For the FASTEM, you can add a new VA on the FastEMMainGUIData, or on a new subclass of MicrosopyGUIData that would be used by FasteEMMainTab.